### PR TITLE
Fixed issue #242

### PR DESCRIPTION
**Pull request by umutondersu**
Changed RemoteBuffer to RemoteState on all paths

### CODE REVIEW

File: silkrpc/core/evm_executor.hpp [hunk 1]
```diff
@@ -65,7 +65,7 @@ class EVMExecutor {
     const core::rawdb::DatabaseReader& db_reader_;
     const silkworm::ChainConfig& config_;
     asio::thread_pool& workers_;
-    state::RemoteBuffer buffer_;
+    state::RemoteState buffer_;
```
Comment (canepat) [inline silkrpc/core/evm_executor.hpp:68]: After renaming the variable type `RemoteBuffer` as `RemoteState`, also the variable name `buffer_` should be renamed `remote_state_`

File: silkrpc/core/remote_buffer.hpp [hunk 2]
```diff
@@ -124,10 +124,10 @@ class RemoteBuffer : public silkworm::State {
 
 private:
     asio::io_context& io_context_;
-    AsyncRemoteBuffer async_buffer_;
+    AsyncRemoteState async_buffer_;
```
Comment (canepat) [inline silkrpc/core/remote_buffer.hpp:127]: Same as above: probably `async_state_` is now better than `async_buffer_`

File: silkrpc/core/remote_buffer_test.cpp [hunk 3]
```diff
@@ -89,7 +89,7 @@ TEST_CASE("async remote buffer", "[silkrpc][core][remote_buffer]") {
         asio::io_context io_context;
         MockDatabaseReader db_reader;
         const uint64_t block_number = 1'000'000;
-        AsyncRemoteBuffer arb{io_context, db_reader, block_number};
+        AsyncRemoteState arb{io_context, db_reader, block_number};
```
Comment (canepat) [inline silkrpc/core/remote_buffer_test.cpp:92]: Same as above: probably `ars` or `state` could be better now than `arb`

File: silkrpc/core/remote_buffer_test.cpp [hunk 4]
```diff
@@ -116,7 +116,7 @@ TEST_CASE("async remote buffer", "[silkrpc][core][remote_buffer]") {
         MockDatabaseReader db_reader{code};
         const uint64_t block_number = 1'000'000;
         const auto code_hash{0x04491edcd115127caedbd478e2e7895ed80c7847e903431f94f9cfa579cad47f_bytes32};
-        RemoteBuffer remoteBuffer(io_context, db_reader, block_number);
+        RemoteState remoteBuffer(io_context, db_reader, block_number);
```
Comment (canepat) [inline silkrpc/core/remote_buffer_test.cpp:119]: Same as above: probably `remote_state` or `state` could be better now than `remoteBuffer`

**Review by canepat — CHANGES_REQUESTED** [review]
Thanks a lot for contributing! Given that we're changing a couple of type names in this PR, I've just suggested the same changes also in variable names,

**Comment by enriavil1** [comment]
#242

**Review by canepat — APPROVED** [review]

**Comment by canepat** [comment]
> #242

Thanks a lot @enriavil1 for bringing this PR to completion!

**Pull request merged**